openfasoc/generators/glayout/glayout/flow/blocks/strongARM_US/strongARM.py

```diff
@@ -0,0 +1,160 @@
+    #move everything to position
+    for comp, prt, alignment in move_info:
+        alignment = ('c','b') if alignment is None else alignment
+        compref = align_comp_to_port(comp, prt, alignment=alignment)
+        diffPair.add(compref)
+    
+    diffPair.flatten()
+    diffPair.write_gds("./strong.gds")
+    #spice netlist
+    fetL = nmos(pdk, width=2*width, length=length, with_dummy=(True, False))
+    fetR = nmos(pdk, width=2*width, length=length, with_dummy=(False, True))
+    diffPair.info['netlist'] = diffPair_netlist(fetL, fetR)
+    #print(diffPair.info['netlist'].generate_netlist())
+    #print(diffPair.get_ports_list())
+    lvs_result = sky130_mapped_pdk.lvs_netgen(diffPair,'diffPair')
+    return diffPair
+
+diffPair(sky130_mapped_pdk,2,0.4).show()
+
+
+def cross_coupled_load(pdk: MappedPDK, ccinv_col):
+    cross_coupled_load=Component(name="cross_coupled_load")
+    cross_couple = generic_4T_interdigitzed(pdk, numcols=ccinv_col, top_row_device="pfet", bottom_row_device="nfet", length=0.4)
+    cross_couple_ref = prec_ref_center(cross_couple)
+    cross_coupled_load.add(cross_couple_ref)
+    cross_coupled_load.add_ports(cross_couple_ref.get_ports_list(), prefix="cross_couple_")
+    cross_coupled_load << smart_route(pdk,cross_coupled_load.ports["cross_couple_top_A_source_E"],cross_coupled_load.ports["cross_couple_top_B_source_E"],cross_couple_ref,cross_coupled_load)
+    cross_coupled_load << smart_route(pdk,cross_coupled_load.ports["cross_couple_bottom_A_source_E"],cross_coupled_load.ports["cross_couple_bottom_B_source_E"],cross_couple_ref,cross_coupled_load)
+    cross_coupled_load << smart_route(pdk,cross_coupled_load.ports["cross_couple_top_A_drain_E"],cross_coupled_load.ports["cross_couple_top_B_gate_E"],cross_couple_ref,cross_coupled_load)
+    cross_coupled_load << smart_route(pdk,cross_coupled_load.ports["cross_couple_bottom_A_drain_E"],cross_coupled_load.ports["cross_couple_bottom_B_gate_E"],cross_couple_ref,cross_coupled_load)
+    cross_coupled_load << smart_route(pdk,cross_coupled_load.ports["cross_couple_top_B_drain_E"],cross_coupled_load.ports["cross_couple_top_A_gate_E"],cross_couple_ref,cross_coupled_load)
+    cross_coupled_load << smart_route(pdk,cross_coupled_load.ports["cross_couple_bottom_B_drain_E"],cross_coupled_load.ports["cross_couple_bottom_A_gate_E"],cross_couple_ref,cross_coupled_load)
+    cross_coupled_load << smart_route(pdk,cross_coupled_load.ports["cross_couple_top_B_gate_E"],cross_coupled_load.ports["cross_couple_bottom_B_gate_E"],cross_couple_ref,cross_coupled_load)
+    cross_coupled_load << smart_route(pdk,cross_coupled_load.ports["cross_couple_top_A_gate_W"],cross_coupled_load.ports["cross_couple_bottom_A_gate_W"],cross_couple_ref,cross_coupled_load)
+    return cross_coupled_load
+
+#cross_coupled_load(gf180_mapped_pdk,2,0.5).show()
+
+def mystrongARM(pdk: MappedPDK, diffp_w, diffp_l, ccinv_col, clk_fing, reset_w, reset_l):
```

Comment:
add more parameterization here, for the diffpair transistors' length, width,etc. 
also add options for taprings, welltaps and such